elastic/integrations
# Add docker_logs datastream for docker container logs collection

## What does this PR do?
Add `docker_logs` data_stream in `docker` package so as to support containers' logs collection using the `docker` dynamic provider.


## Related issues
- Relates https://github.com/elastic/integrations/issues/3163


## Screenshots

![Pr1](https://user-images.githubusercontent.com/11754898/203764517-5061460a-e586-433c-87d4-bd2c91377d5a.png)
![Pr3](https://user-images.githubusercontent.com/11754898/203767614-0007a0c7-7c0e-4403-a22b-1ec6e3475726.png)
![Pr2](https://user-images.githubusercontent.com/11754898/203764495-6b4f8813-ac08-4749-9d2d-f70b33b2de70.png)
![PRScreenshot](https://user-images.githubusercontent.com/11754898/203764004-f3e860a9-2738-48b8-9675-4242277e0f3d.png)

## How to test this PR locally

0. Make sure that you have access to `/var/run/docker.sock` and `/var/lib/docker/containers`. If needed change the permissions for example with `sudo chmod -R 777 /var/run/docker.sock && sudo chmod -R 777 /var/lib/docker/containers`.
1. Edit the docker-compose file that is used by `elastic-package` to bring the local stack up:
In Elastic Agent section add the socket and the log dirs as volume mounts:
```
volumes:
    - "../certs/ca-cert.pem:/etc/ssl/certs/elastic-package.pem"
    - "/var/run/docker.sock:/var/run/docker.sock:ro"
    - "/var/lib/docker/containers:/var/lib/docker/containers:ro"
```
2. Bring stack up with `elastic-package stack up -d -v --version=8.5.0`
3. Add the docker package from fleet UI and make sure that logs from containers are being collected.
